exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/spill/RecordBatchSizer.java

```diff
@@ -70,52 +72,90 @@
      */
 
     public final int estSize;
+
+    /**
+     * Number of times the value here (possibly repeated) appears in
+     * the record batch.
+     */
+
     public final int valueCount;
-    public final int entryCount;
-    public final int dataSize;
-    public final int estElementCount;
+
+    /**
+     * Number of times the entries of this column appears. If this is a
+     * scalar, the entry count is the same as the value count. If this
```

Comment:
The comment is poorly worded. Meant "array of scalar" vs. "Array of variable-width". There is also "array of array of scalar | variable width"... Please check if the revised comment makes this any clearer. (It is hard to explain without first explaining the whole column data model...)